sdks/go/client.go

```diff
@@ -93,12 +88,20 @@ func (c *Client) NewWorker(name string, options ...WorkerOption) (*Worker, error
 
 	// Register regular workflows with non-durable worker
 	for _, workflow := range regularWorkflows {
-		req, _, _, onFailureFn := workflow.Dump()
+		req, regularActions, _, onFailureFn := workflow.Dump()
 		err := nonDurableWorker.RegisterWorkflowV1(req)
 		if err != nil {
 			return nil, err
 		}
 
+		// Register all regular task actions
+		for _, namedFn := range regularActions {
```

Comment:
Note that this still isn't exactly correct, since we're only looking at `regularWorkflows`, but a workflow can have both durable and non-durable tasks. So we're failing to register any non-durable tasks which are on a "durable" workflow. We shouldn't think of workflows as durable or non-durable, we should think of tasks as durable and non-durable.